javascript/ql/lib/semmle/javascript/Actions.qll

```diff
@@ -0,0 +1,322 @@
+/**
+ * Libraries for modeling GitHub Actions workflow files written in YAML.
+ * See https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions.
+ */
+
+import javascript
+
+/**
+ * Libraries for modeling GitHub Actions workflow files written in YAML.
+ * See https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions.
+ */
+module Actions {
+  /** A YAML node in a GitHub Actions workflow file. */
+  private class Node extends YAMLNode {
+    Node() {
+      this.getLocation()
+          .getFile()
+          .getRelativePath()
+          .regexpMatch("(^|.*/)\\.github/workflows/.*\\.yml$")
+    }
+  }
+
+  /**
+   * A YAML node that may contain sub-nodes.
+   *
+   * Actions are quite flexible in parsing YAML.
+   *
+   * For example:
+   * ```
+   * on: pull_request
+   * ```
+   * and
+   * ```
+   * on: [pull_request]
+   * ```
+   * and
+   * ```
+   * on:
+   *   pull_request:
+   * ```
+   *
+   * are equivalent.
+   */
+  class MappingOrSequenceOrScalar extends YAMLNode {
```

Comment:
Should this be in the standard library instead of in Actions.qll.
Actions can't be the only place where this liberal parsing is done.